amr-wind/incflo_advance.cpp

```diff
@@ -268,7 +282,11 @@ void incflo::ApplyPredictor(bool incremental_projection)
     }
 
     // Extrapolate and apply MAC projection for advection velocities
-    icns().pre_advection_actions(amr_wind::FieldState::Old);
+    if (inonlin == 0) {
```

Comment:
Store the field state with a ternary expression then call pre_advection_actions with that variable.

Reply:
I removed the if statements and added a ternary expression the same way it was done for `icns()compute_advection_term()`.